### PR TITLE
 BaSP-MR-58: Activity - Add and Divide Components

### PR DESCRIPTION
**Pull request by DelfiLT**
Features:
- Divided the index in three different components, index, Table and Form.
- Added style to be equal than the table from SuperAdmin Figma wich is the one we are all following (See Picture)
- Added my teammates components.
- Added routes for activity create and add

Figma: 
image
image

Activities Style: 
image
image
image

### CODE REVIEW

**Review by ukiweskamp — COMMENTED** [review]
From what I understood from the planning, I think you should just have Index.jsx with the table and also a form component.
You should remove src/Components/Activities/Table/index.jsx

**Comment by DelfiLT** [comment]
> From what I understood from the planning, I think you should just have Index.jsx with the table and also a form component. You should remove src/Components/Activities/Table/index.jsx

The Table index is for the Table logic, it's not a new page, as I understood you have one page for the table and one for the form, my table is being called in the index to have a better component organization, it doesn't render a new page as well

**Review by SantiH97 — COMMENTED** [review]
As a user its not clear to me what the status checkbox changes, upon reading the code I'm just as lost. Perhaps you could change its name to something more descriptive.
The add or edit button on the form should redirect you to the activity page.
I find it weird to have the success modal on the top right corner but Im guessing thats an issue with the modal itself.

**Comment by SantiH97** [comment]
I also noticed the backgroud image repeats. I suggest adding 
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
to the ".formContainer" class

**Comment by DelfiLT** [comment]
> As a user its not clear to me what the status checkbox changes, upon reading the code I'm just as lost. Perhaps you could change its name to something more descriptive. The add or edit button on the form should redirect you to the activity page. I find it weird to have the success modal on the top right corner but Im guessing thats an issue with the modal itself.

The checkbox status its to set the Activity state from active to inactive, due to the isActive prop but I'll change this name to be more specific, and I tried to make the page turn back, but if I do that I cant trigger the modal. And about the modal it isn't up to me to choose where it appears, it's already in the modal logic.

File: src/Components/Activities/Form/index.jsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+            />
+          </div>
+          <div className={styles.btns}>
+            <Link to="/activities">
+              <Button text={'Cancel'} type={'white'} />
+            </Link>
+            <Button
+              text={location.pathname.includes('edit') ? 'Edit' : 'Add'}
+              type={'add'}
+              clickAction={sendActivity}
```
Comment (Dannylez) [inline src/Components/Activities/Form/index.jsx:145]: When you create or edit an activity, does it automatically redirect to the list of activities? I can't find the logic for that

Comment (DelfiLT) [reply src/Components/Activities/Form/index.jsx:145]: Im on it!

**Comment by SantiH97** [comment]
I think the isActive part of the activity object shouldn't be a user-mutable flag, it seems to me to be more of an admin or super admin way of disabling an established activity without deleting it to be able to bring it back at a moments notice. In other words a PUT delete flag. Since the server isn't really set up to handle that functionality but it still requires the flag my suggestion is that you set that flag to true outside of the users view, hardcoding it into the activity object itself

**Comment by SantiH97** [comment]
What if the success modal shows up after the screen changes back to the previous table?

**Comment by DelfiLT** [comment]
> I think the isActive part of the activity object shouldn't be a user-mutable flag, it seems to me to be more of an admin or super admin way of disabling an established activity without deleting it to be able to bring it back at a moments notice. In other words a PUT delete flag. Since the server isn't really set up to handle that functionality but it still requires the flag my suggestion is that you set that flag to true outside of the users view, hardcoding it into the activity object itself

But I don't think what are we doing here is for the users to use, create and edit activities can only be done by admins, so if your are going to use this entity from the point of view of the admin, you should be able to active or disactive an activity, this feature isnt something that the user uses

**Comment by DelfiLT** [comment]
> What if the success modal shows up after the screen changes back to the previous table?

I've updated that in the last commit

**Comment by SantiH97** [comment]
That does make sense, when you're right you're right but in which context would anyone need to make an inactive activity? Maybe I'm overthinking this but wouldnt it make more sense if it was only an edit functionality?

**Comment by DelfiLT** [comment]
> That does make sense, when you're right you're right but in which context would anyone need to make an inactive activity? Maybe I'm overthinking this but wouldnt it make more sense if it was only an edit functionality?

From my perspective when you create an activity it should be inactive by default, imagine an admin wants to add a new activity but he doesn't want it to be shown to the users, by default it would be inactive, but maybe the admin wants to create it already active, so I think this feature has to be in both cases, when you create and when you edit an activity. And also, if you go to the server, the isActive property is a required file to create one, so if you don't have it, it should give you an error from the backend.

**Review by ukiweskamp — APPROVED** [review]

**Review by SantiH97 — APPROVED** [review]
Ill leave that last comment as a mere suggestion, food for thought, everything else looks great, keep up the good work  :)

**Pull request merged**